app/cli/cmd/attestation_add.go

```diff
@@ -192,3 +204,103 @@ func displayMaterialInfo(status *action.AttestationStatusMaterial, policyEvaluat
 	mt.Render()
 	return nil
 }
+
+// unrecognizedSchemeError is an error type for when a URL scheme is not recognized out of
+// the supported ones.
+type unrecognizedSchemeError struct {
+	Scheme string
+}
+
+func (e *unrecognizedSchemeError) Error() string {
+	return fmt.Sprintf("loading URL: unrecognized scheme: %s", e.Scheme)
+}
+
+// loadFileOrURL tries to load a file or URL from the given path.
+// If the path starts with "http://" or "https://", it will try to load the file from the URL and save it
+// in a temporary file. It will return the path to the temporary file.
+// If the path is an actual file path, it will return the filepath
+func loadFileOrURL(resourcePath string) (string, error) {
+	if _, err := os.Stat(resourcePath); err == nil {
+		return resourcePath, nil
+	}
+
+	// Try to load the resource from a URL
+	raw, err := loadResource(resourcePath)
+	if err != nil {
+		// If the error is an unrecognized scheme error, it means the path is not a URL
+		// and we should return the path as is
+		var uerr *unrecognizedSchemeError
+		if errors.As(err, &uerr) {
+			return resourcePath, nil
+		}
+		return "", fmt.Errorf("loading resource: %w", err)
+	}
+
+	// If the resource is loaded from a URL, save it in a temporary file
+	return createTempFile(resourcePath, raw)
+}
+
+func loadResource(resourcePath string) ([]byte, error) {
+	parts := strings.SplitAfterN(resourcePath, "://", 2)
+	// If the path does not contain a scheme, it is considered a file path
+	if len(parts) != 2 {
+		return nil, &unrecognizedSchemeError{Scheme: parts[0]}
+	}
+
+	switch parts[0] {
+	case "http://", "https://":
+		return loadFromURL(resourcePath)
+	case "env://":
+		return loadFromEnv(parts[1])
+	default:
+		return nil, &unrecognizedSchemeError{Scheme: parts[0]}
+	}
+}
+
+// loadFromURL loads the content of a URL and returns it as a byte slice.
+func loadFromURL(url string) ([]byte, error) {
```

Comment:
What about moving all these methods to its own file in this package? maybe "resource_loader.go" or similar